packages/peregrine/src/hooks/useQueryResult.js

```diff
@@ -0,0 +1,42 @@
+import { useReducer } from 'react';
+
+const initialState = {
+    data: null,
+    error: null,
+    loading: false
+};
+
+const reducer = (state, { payload, type }) => {
+    switch (type) {
+        case 'set data': {
```

Comment:
I like to export these sort of variables as consts for use elsewhere.

Reply:
Re-using a reducer? That's a new one; I can't think of when we'd do that. 🤔

Reply:
Ah, I misunderstood this comment originally—I thought you were suggesting to export `reducer`, but instead you were referring to the action type, `"set data"`.

@supernova-at and I were discussing this in another thread in this PR, but action types here are a bit different from action types in Redux.

- They're not global, so they don't need to be namespaced
- There's no logical place to export/import them from
- They're basically just members of an enum

That said, I understand the concern, so I've adjusted the hook to return [bound action creators](https://redux.js.org/basics/actions#action-creators), such as `setLoading()`, in addition to `dispatch()`. We already use this pattern in Venia, and this way consumers won't have to import anything **or** look up action types.